### ISSUE TITLE
Creator does not ensure the subset has only alphabetical characters

### ISSUE THREAD

**Issue by BigRoy** [getavalon/core#289]
Currently the instance creator does not ensure the subset has a name consisting only of alphabetical characters, any symbol, space or special characters seems to be allowed in the Creator.

Is this ok to be generic and abstract? Or should it never allow so to begin with? 

For example, should a space even be allowed? Or more specific, what characters _should_ be?

**Comment by mottosso** [comment]
It should be the case that bad names cannot make it into the database, due to [this](https://github.com/getavalon/core/blob/master/avalon/schema/asset-2.0.json#L39)

Currently the allowed characters are `^[a-zA-Z0-9_.]*$`. We should get that filter into the UI as well.

**Comment by BigRoy** [comment]
> It should be the case that bad names cannot make it into the database, due to this.

Perfect. Thanks for pointing that out, good to know. That should definitely find its way into the UI too.

**Comment by davidlatwe** [comment]
On UX part, this could be related to #179, and with the implementation in #444, use reddish `border-color` to indicate invalid subset name.

 Will submit another PR to resolve this, after #444 is merged.
